### ISSUE TITLE
S3 TransferManager MultiFileDownload fails for filenames < 3 characters and size > 5MB

### ISSUE THREAD

**Issue by kierarad** [aws/aws-sdk-java#906]
When we call TransferManager downloadDirectory() to retrieve a files from S3 which require multipart download (over 5MB) with file names that are less than 3 characters we receive the following exception:

> com.amazonaws.AmazonClientException: Unable to complete transfer: java.lang.IllegalArgumentException: Prefix string too short

  

Here is the corresponding stacktrace. 

```
Caused by: java.util.concurrent.ExecutionException: java.lang.IllegalArgumentException: Prefix string too short
  at java.util.concurrent.FutureTask.report(FutureTask.java:122)
  at java.util.concurrent.FutureTask.get(FutureTask.java:192)
  at com.amazonaws.services.s3.transfer.DownloadCallable.combineFiles(DownloadCallable.java:203) 
  at com.amazonaws.services.s3.transfer.DownloadCallable.downloadInParallel(DownloadCallable.java:193) 
  at com.amazonaws.services.s3.transfer.DownloadCallable.call(DownloadCallable.java:131) 
  at com.amazonaws.services.s3.transfer.DownloadCallable.call(DownloadCallable.java:51) 
  at java.util.concurrent.FutureTask.run(FutureTask.java:266)
  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
  ... 1 common frames omitted
Caused by: java.lang.IllegalArgumentException: Prefix string too short
  at java.io.File.createTempFile(File.java:2001)
  at com.amazonaws.services.s3.transfer.internal.DownloadPartCallable.call(DownloadPartCallable.java:49) 
  at com.amazonaws.services.s3.transfer.internal.DownloadPartCallable.call(DownloadPartCallable.java:31) 
  ... 4 common frames omitted
```

**Comment by varunnvs92** [comment]
Thanks for reporting this issue. It will be fixed in tomorrow's release.

**Comment by varunnvs92** [comment]
The issue is fixed in [v1.11.49](https://aws.amazon.com/releasenotes/2981276408528428).

Thanks for notifying us about this issue.

**Comment by kierarad** [comment]
Thank you
